accepted/00097-recurring-actions.md

```diff
@@ -0,0 +1,193 @@
+
+## XMLRPC / HTTP API
+
+We already have a `RecurringActionHandler` that can be expanded to support other action types.
+
+I see two possibilities two implement new functions:
+
+1) For each type implement individiual CRUD function that states the action type in its name. E.g. `RecurringActionHandler.createCustomState(...)`. This is essentially just a wrapper that calls the generic `create` function with the action type as a parameter.
+
+2) Have one CRUD function for all the types and pass the action type as an argument.
+
+Option 1) could be more intuitive to use and its function would be clear just by reading its name, but add a lot of extra API endpoints that all call the same underlying base function, so option 2) would be less overhead.
+
+In the end I don't have a strong opinion which approach to take.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## SSM
+
+There are two ways recurring actions could work with SSM. However both come with significant downsides.
+
+1) Creating a recurrent action via SSM will schedule a `MinonRecurringAction` for every selected minion. This would schedule one `Taskomatic` job for every minion that would be scheduled to run at the same time.
+Needless to say this would set us up for disaster and overwhelm `Taskomatic` if done with more than a few minions.
+
+2) Create a new `SSMRecurringAction` entity that would remember the Minions selected for the schedule by creating a db table e.g. called `RecurringSSMMinions` with a one-to-many relationship with the SSM entity. This way entries would be updated properly if a system is deleted (given it's configured properly with a foreign key and delete cascade to the minion).
+This way we could have one `Taskomatic` job that would perform the action for all the minions it was created with.
+
+Each SSM action could be displayed in the minions recurring action list. Updating/Editing the action on minion level would either be forbidden or update it for all the minions it was created for. 
+
+However adding/removing minions from an existing SSM schedule would require implementation of dedicated UI to support.
+
+In the end I'd advise against supporting recurring actions with SSM. We would map something that's designed to be not persistent (SSM selection) to something that is (Recurring Action schedule). 
+
+# Alternatives
+[alternatives]: #alternatives
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+**Question:** With the ability to schedule workflows on recurrent basis, the load on Taskomatic is expected to increase. Especially if there is a loot of schedules with frequent recurrance. Some actions the customers used to schedule manually are going to be perfomed automatically and thus most likely more often.  Is taskomatic up for the task?
```

Comment:
this needs to be tested. Specially the ones that will bring load to salt-master and the CLM promote/build, which are heavy.